src/BaseOfTalents/DAL/Services/ReportService.cs

```diff
@@ -143,17 +159,38 @@ private int GetAddedCandedatesCount(DateTime startDate, DateTime endDate, int us
             var vacanciesGroupedByLocations = states.Select(x => x.Vacancy).GroupBy(x => x.ResponsibleId).Select(x => x.First())
                 .GroupBy(x => x.Cities.First().Id);
 
-
-
             var result = vacanciesGroupedByLocations.Select(x => new LocationDailyVacanciesReportDTO
             {
                 LocationId = x.Key,
-                DailyVacanciesStatisticsInfo = x.Select(y => CreateDailyVacanciesReport(y.ResponsibleId, y.Responsible.FirstName, y.Responsible.LastName, states)).ToList()
+                VacanciesStatisticsInfo = x.Select(y => CreateDailyVacanciesReport(y.ResponsibleId, y.Responsible.FirstName, y.Responsible.LastName, states)).ToList()
             });
 
-            return result;
+            var distinct = locationIds.Except(vacanciesGroupedByLocations.Select(x => x.Key)).ToList();
+            var resultContainsEmptyObjects = new List<LocationDailyVacanciesReportDTO>();
+            if (distinct.Count != 0)
+            {
+                var emptyObjectResult = GetReportResultContainsEmptyObjects<LocationDailyVacanciesReportDTO>(distinct);
+                resultContainsEmptyObjects = result.ToList();
+                resultContainsEmptyObjects.AddRange(emptyObjectResult);
+                return resultContainsEmptyObjects;
+            } else {
+                return result;
+            };
         }
 
+        private List<T> GetReportResultContainsEmptyObjects<T> (List<int> distinct) where T : LocationVacancyReportDTO, new()
+        {
+            List<T> resultWithEmptyObjects = new List<T>();
```

Comment:
you can do easily: **var result = distinct.Select(x => new T() { LocationId: x });**